blocks/query-heading/render.php

```diff
@@ -2,36 +2,31 @@
 /**
  * All of the parameters passed to the function where this file is being required are accessible in this scope:
  *
- * @var array<mixed> $attributes The array of attributes for this block.
+ * @var array{
+ *   level?: int,
+ *   override?: string,
+ * } $attributes The array of attributes for this block.
  * @var string       $content    Rendered block output. ie. <InnerBlocks.Content />.
  * @var WP_Block     $block      The instance of the WP_Block class that represents the block being rendered.
  *
  * @package wp-curate
  */
 
-$wp_curate_heading = '';
+use Alley\WP\WP_Curate\WP_Curate;
 
-if ( isset( $block->context['heading']['source'] ) ) :
-	if ( 'custom' === $block->context['heading']['source'] && isset( $block->context['heading']['custom'] ) ) :
-		$wp_curate_heading = $block->context['heading']['custom'];
-	endif;
+$wp_curate_heading     = '';
+$wp_curate_heading_tag = isset( $attributes['level'] ) ? "h{$attributes['level']}" : 'h2';
 
-	if (
-		'termId' === $block->context['heading']['source']
-		&& isset( $block->context['heading']['termId'], $block->context['heading']['taxonomy'] )
-	) :
-		$wp_curate_term = get_term( $block->context['heading']['termId'], $block->context['heading']['taxonomy'] );
-
-		if ( $wp_curate_term instanceof WP_Term ) :
-			$wp_curate_heading = html_entity_decode( $wp_curate_term->name );
-		endif;
-	endif;
+if ( ! empty( $attributes['override'] ) ) : // Static heading override.
+	$wp_curate_heading = $attributes['override'];
+elseif ( isset( $block->context['curation'] ) ) : // Dynamic heading based on block context.
+	$wp_curate_heading = WP_Curate::get_query_heading( $block->context['curation'] );
+endif;
 
-	if ( is_string( $wp_curate_heading ) && '' !== $wp_curate_heading ) :
-		?>
-			<h2 <?php echo wp_kses_data( get_block_wrapper_attributes( [ 'class' => 'wp-block-heading' ] ) ); ?>>
-				<?php echo esc_html( $wp_curate_heading ); ?>
-			</h2>
-		<?php
-	endif;
+if ( is_string( $wp_curate_heading ) && '' !== $wp_curate_heading ) :
+	?>
+		<<?php echo esc_attr( $wp_curate_heading_tag); ?> <?php echo wp_kses_data( get_block_wrapper_attributes( [ 'class' => 'wp-block-heading' ] ) ); ?>>
```

Comment:
does this class solve the concern of heading styles not getting loaded? I know you brought that up the other day...

Reply:
No, I don't think it does. This was from The Wrap, but I actually don't think that class is needed.